src/workflow/StreamlitUI.py

```diff
@@ -97,18 +106,33 @@ def upload_widget(
                 if st.form_submit_button(
                     f"Copy **{name}** files from local folder", use_container_width=True
                 ):
-                    # raw string for file paths
-                    if not any(Path(local_dir).glob(f"*.{file_type}")):
+                    files = []
+                    local_dir = Path(
+                        local_dir
+                    ).expanduser()  # Expand ~ to full home directory path
+
+                    for ft in file_types:
+                        # Search for both files and directories with the specified extension
+                        for path in local_dir.iterdir():
+                            if path.is_file() and path.name.endswith(f".{ft}"):
+                                files.append(path)
+                            elif path.is_dir() and path.name.endswith(f".{ft}"):
+                                files.append(path)
+
+                    if not files:
                         st.warning(
-                            f"No files with type **{file_type}** found in specified folder."
+                            f"No files with type **{', '.join(file_types)}** found in specified folder."
                         )
                     else:
-                        # Copy all mzML files to workspace mzML directory, add to selected files
-                        files = list(Path(local_dir).glob("*.mzML"))
                         my_bar = st.progress(0)
                         for i, f in enumerate(files):
                             my_bar.progress((i + 1) / len(files))
-                            shutil.copy(f, Path(files_dir, f.name))
+                            if not symlink:
+                                shutil.copy(f, Path(files_dir, f.name))
+                            else:
+                                symlink_target = Path(files_dir, f.name)
+                                if not symlink_target.exists():
+                                    os.symlink(f, symlink_target)
```

Comment:
Unfortunately, this does not work under windows (except when adjusting security policies or running as an administrator). Since this is currently our main use case for local deployments, we would need another solution here.

Reply:
See [this](https://stackoverflow.com/questions/26787872/how-to-create-symlinks-in-windows-using-python) StackOverflow thread. I also tested it on my local machine.